File: customer-service/src/main/java/com/wks/servicemarketplace/customerservice/adapters/auth/DefaultSecurityContext.java
package com.wks.servicemarketplace.customerservice.adapters.auth;

import com.wks.servicemarketplace.common.auth.Authentication;

import javax.ws.rs.core.SecurityContext;
import java.security.Principal;

public class DefaultSecurityContext implements SecurityContext {

    private final Authentication authentication;
    private final boolean isHttps;
    private final String authorizationScheme;

    public DefaultSecurityContext(Authentication authentication, boolean isHttps, String authorizationScheme) {

        this.authentication = authentication;
        this.isHttps = isHttps;
        this.authorizationScheme = authorizationScheme;
    }

    @Override
    public Principal getUserPrincipal() {
        return authentication;
    }

    @Override
    public boolean isUserInRole(String role) {
        return authentication.hasRole(role);
    }

    @Override
    public boolean isSecure() {
        return isHttps;
    }

    @Override
    public String getAuthenticationScheme() {
        return authorizationScheme;
    }

    @Override
    public String toString() {
        return "DefaultSecurityContext{" +
                "user=" + authentication +
                ", isHttps=" + isHttps +
                ", authorizationScheme='" + authorizationScheme + '\'' +
                '}';
    }
}

File: service-provider-service/src/main/kotlin/com/wks/servicemarketplace/serviceproviderservice/adapters/auth/DefaultSecurityContext.kt
package com.wks.servicemarketplace.serviceproviderservice.adapters.auth

import com.wks.servicemarketplace.common.auth.Authentication
import com.wks.servicemarketplace.common.auth.Permission
import java.lang.Exception
import java.security.Principal
import javax.ws.rs.core.SecurityContext

class DefaultSecurityContext(private val authentication: Authentication,
                             private val isHttps: Boolean,
                             private val authorizationScheme: String?) : SecurityContext {

    override fun getUserPrincipal(): Principal? {
        return authentication
    }

    override fun isUserInRole(role: String): Boolean {
        return try {
            authentication.hasRole(Permission.valueOf(role))
        } catch (e: Exception) {
            false
        }
    }

    override fun isSecure(): Boolean {
        return isHttps
    }

    override fun getAuthenticationScheme(): String? {
        return authorizationScheme
    }

    override fun toString(): String {
        return "DefaultSecurityContext{" +
                "user=" + authentication +
                ", isHttps=" + isHttps +
                ", authorizationScheme='" + authorizationScheme + '\'' +
                '}'
    }
}